iothub_client/src/iothubtransporthttp.c

```diff
@@ -1540,199 +1744,76 @@ static void DoEvent(HTTPTRANSPORT_HANDLE_DATA* handleData, HTTPTRANSPORT_PERDEVI
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair");
-                                        goOn = false;
-                                    }
-                                }
-
-                                corrId = IoTHubMessage_GetCorrelationId(message->messageHandle);
-                                if (goOn && corrId != NULL)
-                                {
-                                    if (HTTPHeaders_ReplaceHeaderNameValuePair(clonedEventHTTPrequestHeaders, IOTHUB_CORRELATION_ID, corrId) != HTTP_HEADERS_OK)
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair");
-                                        goOn = false;
-                                    }
-                                }
-
-                                // Codes_SRS_TRANSPORTMULTITHTTP_09_001: [ If the IoTHubMessage being sent contains property `content-type` it shall be added to the HTTP headers as "iothub-contenttype":"value". ]
-                                userDefinedContentType = IoTHubMessage_GetContentTypeSystemProperty(message->messageHandle);
-                                if (goOn && userDefinedContentType != NULL)
-                                {
-                                    if (HTTPHeaders_ReplaceHeaderNameValuePair(clonedEventHTTPrequestHeaders, IOTHUB_CONTENT_TYPE_D2C, userDefinedContentType) != HTTP_HEADERS_OK)
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair (content-type)");
-                                        goOn = false;
-                                    }
-                                }
-
-                                // Codes_SRS_TRANSPORTMULTITHTTP_09_002: [ If the IoTHubMessage being sent contains property `content-encoding` it shall be added to the HTTP headers as "iothub-contentencoding":"value". ]
-                                contentEncoding = IoTHubMessage_GetContentEncodingSystemProperty(message->messageHandle);
-                                if (goOn && contentEncoding != NULL)
-                                {
-                                    if (HTTPHeaders_ReplaceHeaderNameValuePair(clonedEventHTTPrequestHeaders, IOTHUB_CONTENT_ENCODING_D2C, contentEncoding) != HTTP_HEADERS_OK)
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair (content-encoding)");
-                                        goOn = false;
-                                    }
-                                }
-
-                                if (!goOn)
+                                if (set_system_properties(message, clonedEventHTTPrequestHeaders) != 0)
```

Comment:
>set_system_properties [](start = 36, length = 21)

Nit - and I get I'm adding a lot of these given this is a good change as is :).  but this and some of other blocks could be moved to a

if (set_messag_propes() == fail)
{  LogFailure() ; }
else if (set_system_prorites == )
{...
}